src/dropdown/Dropdown.js

```diff
@@ -24,34 +22,46 @@ export default class Dropdown extends React.Component {
     this.logWarnings();
   }
 
-  getGeneralProps = () => omit(this.props, Object.keys(PropTypes));
-
   logWarnings = () => {
     if (this.props.children) {
       warn('You\'re passing children. Not expected behaviour');
     }
   };
 
-  getValueLinkProps = () => {
-    if (this.props.valueLink) {
-      return {
-        value: this.props.valueLink.value,
-        onChange: this.props.valueLink.requestChange
-      };
+  getValue = () => (
+    this.props.valueLink ? this.props.valueLink.value : this.props.value
+  );
+
+  valueToOption = (value, options) => {
+    if (t.String.is(value) || t.Number.is(value)) {
+      return find(options, { value });
     }
+    return value;
   };
 
-  getClassName = () => cx(this.props.className, themes[this.props.theme]);
-
-  render() {
-    // The order is important: props may override previous ones
-    return (
-      <Select
-        {...this.getGeneralProps()}
-        {...this.getValueLinkProps()}
-        {...this.getClassName()}
-      />
-    );
+  _onChange = () => this.props.valueLink ? this.props.valueLink.requestChange : this.props.onChange;
+
+  _onBlur = () => this.forceUpdate();
+
+  getLocals() {
+    const {
+      className,
+      options,
+      valueLink,
+      ...props
+    } = this.props;
+
+    return {
+      ...props,
+      options,
+      className: cx('dropdown', className),
+      value: this.valueToOption(this.getValue(), options),
+      onChange: this._onChange(),
```

Comment:
as pointed in the comment above this should be either `this._onChange` or `this.getOnChange()`